### PR TITLE
feat(github): Add support for Draft PRs

### LINKED ISSUE #5850: Support create as Draft PR in Github
**What would you like Renovate to be able to do?**
Has configuration options to be able to create as draft PR.

**Describe the solution you'd like**
if configuration like `draftPR` as boolean is `true` then renovate-bot just create PR as draft in github. (default value should be `false`)

**Misc**
Github API for create pull request as draft is already provided https://developer.github.com/v3/pulls

### PR DESCRIPTION
**Pull request by tanettrimas**
This PR implements the ability to set the config option "draftPR" to true in renovate.json for example. Every PR that is created by Renovate on github is converted to a draft pull request.

See pictures below for it in action.
Skjermbilde 2020-05-30 kl  22 09 55

Skjermbilde 2020-05-30 kl  22 12 47

NOTE: lib/versioning/regex/index.spec.ts and lib/util/regex.spec.ts are currently failing, but I am not quite sure why and how to fix it. Any leads much appreciated :)

Closes #5850

### CODE REVIEW

**Comment by tanettrimas** [comment]
> NOTE: lib/versioning/regex/index.spec.ts and lib/util/regex.spec.ts are currently failing, but I am not quite sure why and how to fix it. Any leads much appreciated :)

Does seem like some of these tests actually passed in CI but failed locally when watching the tests...

**Comment by tanettrimas** [comment]
@rarkins Please have a look :)

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -255,6 +255,10 @@ Add config here if you wish it to apply to Docker package managers Dockerfile an
 
 ## dotnet
 
+## draftPR
+
+If `draftPR` is set to true, the PR created by Renovate is set to a draft. Currently only available in Github.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:260]: This text shouldn't duplicate the text in the config/definitions description. Ideally rewrite this to include a link to draft PRs documentation on github.com

File: lib/platform/github/index.ts [hunk 2]
```diff
@@ -1662,6 +1667,7 @@ export async function createPr({
       url: 'https://github.com/renovatebot/renovate',
     });
   }
+  pr.draft = draftPR;
```
Comment (rarkins) [inline lib/platform/github/index.ts:1670]:
```suggestion
```

This is unnecessary and unused?

File: lib/platform/common.ts [hunk 3]
```diff
@@ -116,6 +116,7 @@ export interface Pr {
   state: string;
   targetBranch?: string;
   title: string;
+  draft?: boolean;
```
Comment (rarkins) [inline lib/platform/common.ts:119]:
```suggestion
```

Unused, so remove

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/platform/github/index.ts (hunk 2 above)
Comment (tanettrimas) [reply lib/platform/github/index.ts:1670]: I am using it in tests and could be used for later. Do you still want me to remove it?

File: lib/platform/common.ts (hunk 3 above)
Comment (tanettrimas) [reply lib/platform/common.ts:119]: I am using it in tests and could be used for later. Do you still want me to remove it?

File: lib/platform/github/index.spec.ts [hunk 4]
```diff
@@ -1575,6 +1575,26 @@ describe('platform/github', () => {
...
+      const pr = await github.createPr({
+        branchName: 'some-branch',
+        prTitle: 'PR draft',
+        prBody: 'This is a result of a draft',
+        labels: null,
+        useDefaultBranch: true,
+        draftPR: true,
+      });
+      expect(pr).toMatchSnapshot();
+      expect(pr.draft).toBe(true);
```
Comment (tanettrimas) [inline lib/platform/github/index.spec.ts:1597]: As you can see, it is used in the new test. If you want, I can remove it from type and test. And just expect the snapshot to match.

File: docs/usage/configuration-options.md (hunk 1 above)
Comment (tanettrimas) [reply docs/usage/configuration-options.md]: Possibly fixed by https://github.com/renovatebot/renovate/pull/6390/commits/f9cb3477367f6951b75665c26347ba55919b5ded

File: lib/platform/github/index.spec.ts (hunk 4 above)
Comment (rarkins) [reply lib/platform/github/index.spec.ts:1597]: I prefer not to change production types if it’s only for the purpose of testing.

Comment (tanettrimas) [reply lib/platform/github/index.spec.ts]: Removed this in latest commit fe70f80

File: lib/platform/common.ts (hunk 3 above)
Comment (tanettrimas) [reply lib/platform/common.ts]: Removed this in latest commit fe70f80

File: lib/platform/github/index.ts (hunk 2 above)
Comment (tanettrimas) [reply lib/platform/github/index.ts]: Removed this in latest commit fe70f80

**Comment by tanettrimas** [comment]
@rarkins Think I have fixed your comments now. Would you have another look perhaps? :D

File: docs/usage/configuration-options.md [hunk 5]
```diff
@@ -255,6 +255,18 @@ Add config here if you wish it to apply to Docker package managers Dockerfile an
...
+
+If you want the PRs created by Renovate to be considered as drafts rather than normal PRs in Github you could add this property to your `renovate.json`:
+
+```json
+{
+  "draftPR": true
+}
+```
+
+Please see [draft pull requests on Github.](https://github.blog/2019-02-14-introducing-draft-pull-requests/) for more information about draft PRs.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:268]:
```suggestion
Please see [draft pull requests on Github](https://github.blog/2019-02-14-introducing-draft-pull-requests/) for more information about draft PRs.
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**